scripts/spider/dedup_checker.py

```
@@ -323,16 +323,27 @@ def _atomic_csv_write(csv_path: str, rows: List[Dict[str, str]]) -> None:
-    either backend.
+    CSV mode is active.  Returns an empty list when no data exists.
     """
     rows: List[Dict[str, str]] = []
 
+    if from_file_only and csv_path and os.path.exists(csv_path):
+        try:
+            with open(csv_path, 'r', newline='', encoding='utf-8') as f:
+                reader = csv.DictReader(f)
+                rows = [dict(r) for r in reader]
+            logger.debug(f"Loaded {len(rows)} dedup records from CSV: {csv_path}")
+        except Exception as e:
```

Ruff (0.15.5)

[warning] Do not catch blind exception: `Exception`

(BLE001)